In [2]:
%load_ext sql

In [3]:
%sql mysql://admin:sql_2022@tickets.ceaw7c57s0vu.us-east-1.rds.amazonaws.com/tickets

### 1. This query sums how many events I have in my Ticketmaster table by genre.

In [6]:
%%sql
SELECT
    COUNT(event_id) AS number_of_events,
    genre
FROM ticketmaster_event_details
GROUP BY genre
ORDER BY number_of_events DESC;

 * mysql://admin:***@tickets.ceaw7c57s0vu.us-east-1.rds.amazonaws.com/tickets
7 rows affected.


[(29, 'Rock'),
 (16, 'Country'),
 (6, 'Pop'),
 (5, 'Comedy'),
 (5, 'Other'),
 (2, 'R&B'),
 (1, 'Hip-Hop/Rap')]

### I have majority rock and country events stored in my Ticketmaster table

### 2. This query compares ticket count for an event to the capacity of the venue by joining the vividseats and venue tables. 

In [8]:
%%sql
SELECT
    vi.name,
    venue,
    ticketCount,
    capacity
FROM vividseats_event_details vi
JOIN venue ve
    ON vi.venue = ve.name
LIMIT 10;

 * mysql://admin:***@tickets.ceaw7c57s0vu.us-east-1.rds.amazonaws.com/tickets
10 rows affected.


[('BTS - Bangtan Boys', 'Allegiant Stadium', 2135, '65,000'),
 ('Bon Jovi', 'Enmarket Arena', 1112, '9,500'),
 ('Jawbreaker', 'House of Blues Chicago', 145, '1,800'),
 ('Clinton Kane', 'Marquis Theater - CO', 30, '250'),
 ('Dave Chappelle', 'Uptown Theater Kansas City', 209, '1,700'),
 ('BTS - Bangtan Boys', 'Allegiant Stadium', 2030, '65,000'),
 ('Garth Brooks', 'Nissan Stadium', 2615, '69,143'),
 ('Yeat', 'Irving Plaza', 103, '1,200'),
 ('Dave Chappelle', 'Vivint Smart Home Arena', 2291, '18,300'),
 ('Dave Chappelle', 'Vivint Smart Home Arena', 2291, '18,300')]

### One of my questions could look into the ratio between ticketCount and capacity

### 3. This query looks at the ticket count and minimum price as number of days until the event changes.

In [12]:
%%sql
SELECT
    name,
    venue,
    ticketCount,
    minPrice,
    eventDate,
    eventDate - currentDate AS days_until_event
FROM vividseats_event_details
ORDER BY eventDate, name, days_until_event DESC
LIMIT 20;

 * mysql://admin:***@tickets.ceaw7c57s0vu.us-east-1.rds.amazonaws.com/tickets
20 rows affected.


[('BTS - Bangtan Boys', 'Allegiant Stadium', 2135, 54, datetime.date(2022, 4, 8), 0),
 ('BTS - Bangtan Boys', 'Allegiant Stadium', 1929, 60, datetime.date(2022, 4, 8), 0),
 ('Bon Jovi', 'Enmarket Arena', 1112, 49, datetime.date(2022, 4, 13), 5),
 ('Bon Jovi', 'Enmarket Arena', 924, 62, datetime.date(2022, 4, 13), 2),
 ('Bon Jovi', 'Enmarket Arena', 885, 49, datetime.date(2022, 4, 13), 1),
 ('Bon Jovi', 'Enmarket Arena', 873, 48, datetime.date(2022, 4, 13), 1),
 ('Jawbreaker', 'House of Blues Chicago', 145, 22, datetime.date(2022, 4, 14), 6),
 ('Jawbreaker', 'House of Blues Chicago', 105, 6, datetime.date(2022, 4, 14), 3),
 ('Jawbreaker', 'House of Blues Chicago', 87, 6, datetime.date(2022, 4, 14), 2),
 ('Jawbreaker', 'House of Blues Chicago', 75, 6, datetime.date(2022, 4, 14), 2),
 ('Jawbreaker', 'House of Blues Chicago', 38, 17, datetime.date(2022, 4, 14), 0),
 ('BTS - Bangtan Boys', 'Allegiant Stadium', 2030, 80, datetime.date(2022, 4, 15), 7),
 ('BTS - Bangtan Boys', 'Allegiant Stadium', 1346, 113, datetime.date(2022, 4, 15), 4),
 ('BTS - Bangtan Boys', 'Allegiant Stadium', 1133, 134, datetime.date(2022, 4, 15), 3),
 ('BTS - Bangtan Boys', 'Allegiant Stadium', 1145, 134, datetime.date(2022, 4, 15), 3),
 ('BTS - Bangtan Boys', 'Allegiant Stadium', 804, 149, datetime.date(2022, 4, 15), 1),
 ('Clinton Kane', 'Marquis Theater - CO', 30, 34, datetime.date(2022, 4, 15), 7),
 ('Clinton Kane', 'Marquis Theater - CO', 38, 29, datetime.date(2022, 4, 15), 4),
 ('Clinton Kane', 'Marquis Theater - CO', 22, 29, datetime.date(2022, 4, 15), 3),
 ('Clinton Kane', 'Marquis Theater - CO', 26, 29, datetime.date(2022, 4, 15), 3)]

### I could look into changes in ticket count and minimum price to gauge demand as the event comes closer to determine whether it's better to sell earlier or later.

### 4. Compare all face value minimum price and resale minimum price to get an idea of whether an event is profitable.

In [17]:
%%sql
SELECT
    ted.name,
    ted.minPrice AS retail_min_price,
    ved.minPrice AS resale_min_price
FROM ticketmaster_event_details ted
JOIN venue
    ON venue.name = ted.venue
JOIN vividseats_event_details ved
    ON venue.name = ved.venue;

 * mysql://admin:***@tickets.ceaw7c57s0vu.us-east-1.rds.amazonaws.com/tickets
307 rows affected.


[('Kane Brown - Blessed & Free Tour', 56, 49),
 ('mxmtoon - rising (the tour) ', 28, 22),
 ('keshi: HELL / HEAVEN TOUR', 35, 22),
 ('Kane Brown - Blessed & Free Tour', 56, 62),
 ('mxmtoon - rising (the tour) ', 28, 6),
 ('keshi: HELL / HEAVEN TOUR', 35, 6),
 ('Dave Matthews Band', 60, 103),
 ('iHeartCountry Festival Presented by Capital One', 29, 103),
 ('Jimmy Buffett', 54, 103),
 ('Netflix Is A Joke Presents: Bill Burr & Friends', 64, 75),
 ('Netflix Is A Joke Presents: Bill Burr & Friends', 64, 75),
 ("Clinton Kane - MAYBE SOMEDAY IT'LL ALL BE OK", 25, 32),
 ('Netflix Is A Joke Presents: Bill Burr: Slight Return', 60, 138),
 ('Pearl Jam', 108, 138),
 ('Pearl Jam', 108, 138),
 ('PAUL McCARTNEY GOT BACK', 30, 127),
 ('Dave Matthews Band', 60, 99),
 ('iHeartCountry Festival Presented by Capital One', 29, 99),
 ('Jimmy Buffett', 54, 99),
 ('Garth Brooks', 92, 84),
 ('mxmtoon - rising (the tour) ', 28, 15),
 ('keshi: HELL / HEAVEN TOUR', 35, 15),
 ('Netflix Is A Joke Presents: Bill Burr: Slight Return', 60, 148),
 ('Pearl Jam', 108, 148),
 ('Pearl Jam', 108, 148),
 ('Dave Matthews Band', 60, 71),
 ('iHeartCountry Festival Presented by Capital One', 29, 71),
 ('Jimmy Buffett', 54, 71),
 ('Garth Brooks', 85, 89),
 ('Garth Brooks', 85, 89),
 ('Dave Matthews Band', 50, 77),
 ('TWICE 4TH WORLD TOUR  III  IN NORTH AMERICA ENCORE', 60, 100),
 ('TWICE 4TH WORLD TOUR  III  IN NORTH AMERICA ENCORE', 60, 100),
 ('Dave Matthews Band', 50, 108),
 ('PAUL McCARTNEY GOT BACK', 40, 404),
 ('Rex Orange County: The Who Cares? Tour', 30, 57),
 ('Luke Combs Stadium Tour', 21, 76),
 ('Garth Brooks', 85, 49),
 ('Garth Brooks', 85, 49),
 ('Kane Brown - Blessed & Free Tour', 47, 57),
 ('Kane Brown - Blessed & Free Tour', 56, 52),
 ('Kane Brown - Blessed & Free Tour', 56, 49),
 ('mxmtoon - rising (the tour) ', 28, 6),
 ('keshi: HELL / HEAVEN TOUR', 35, 6),
 ('Kane Brown - Blessed & Free Tour', 56, 48),
 ('mxmtoon - rising (the tour) ', 28, 6),
 ('keshi: HELL / HEAVEN TOUR', 35, 6),
 ('Dave Matthews Band', 60, 97),
 ('iHeartCountry Festival Presented by Capital One', 29, 97),
 ('Jimmy Buffett', 54, 97),
 ('mxmtoon - rising (the tour) ', 28, 17),
 ('keshi: HELL / HEAVEN TOUR', 35, 17),
 ('Dave Matthews Band', 60, 103),
 ('iHeartCountry Festival Presented by Capital One', 29, 103),
 ('Jimmy Buffett', 54, 103),
 ('Dave Matthews Band', 60, 97),
 ('iHeartCountry Festival Presented by Capital One', 29, 97),
 ('Jimmy Buffett', 54, 97),
 ('Netflix Is A Joke Presents: Bill Burr & Friends', 64, 63),
 ('Netflix Is A Joke Presents: Bill Burr & Friends', 64, 63),
 ('Netflix Is A Joke Presents: Bill Burr & Friends', 64, 63),
 ('Netflix Is A Joke Presents: Bill Burr & Friends', 64, 63),
 ("Clinton Kane - MAYBE SOMEDAY IT'LL ALL BE OK", 25, 55),
 ('PAUL McCARTNEY GOT BACK', 30, 125),
 ('Netflix Is A Joke Presents: Bill Burr: Slight Return', 60, 119),
 ('Pearl Jam', 108, 119),
 ('Pearl Jam', 108, 119),
 ('Garth Brooks', 92, 77),
 ('Dave Matthews Band', 60, 99),
 ('iHeartCountry Festival Presented by Capital One', 29, 99),
 ('Jimmy Buffett', 54, 99),
 ('mxmtoon - rising (the tour) ', 28, 8),
 ('keshi: HELL / HEAVEN TOUR', 35, 8),
 ('Netflix Is A Joke Presents: Bill Burr: Slight Return', 60, 127),
 ('Pearl Jam', 108, 127),
 ('Pearl Jam', 108, 127),
 ('Dave Matthews Band', 60, 72),
 ('iHeartCountry Festival Presented by Capital One', 29, 72),
 ('Jimmy Buffett', 54, 72),
 ('Dave Matthews Band', 50, 73),
 ('Garth Brooks', 85, 83),
 ('Garth Brooks', 85, 83),
 ('TWICE 4TH WORLD TOUR  III  IN NORTH AMERICA ENCORE', 60, 120),
 ('TWICE 4TH WORLD TOUR  III  IN NORTH AMERICA ENCORE', 60, 120),
 ('Dave Matthews Band', 50, 108),
 ('PAUL McCARTNEY GOT BACK', 40, 404),
 ('Rex Orange County: The Who Cares? Tour', 30, 56),
 ('Garth Brooks', 85, 49),
 ('Garth Brooks', 85, 49),
 ('Kane Brown - Blessed & Free Tour', 47, 62),
 ('Luke Combs Stadium Tour', 21, 74),
 ('Kane Brown - Blessed & Free Tour', 56, 48),
 ('Nine Inch Nails', 50, 90),
 ('Nine Inch Nails', 50, 90),
 ('keshi', 30, 108),
 ('C

### Able to join all 3 tables successfully and pull the data I need.

### 5. Sum the amount of venues we have stored grouped by state.

In [19]:
%%sql
SELECT
    COUNT(ID) AS number_of_venues,
    state
FROM venue
GROUP BY state
ORDER BY number_of_venues DESC
LIMIT 10;

 * mysql://admin:***@tickets.ceaw7c57s0vu.us-east-1.rds.amazonaws.com/tickets
10 rows affected.


[(9, 'California'),
 (8, 'Texas'),
 (4, 'Florida'),
 (3, 'Illinois'),
 (3, 'Colorado'),
 (3, 'Tennessee'),
 (3, 'Pennsylvania'),
 (3, 'Massachusetts'),
 (2, 'Ohio'),
 (2, 'Indiana')]

### We could possibly perform some analytics based on state or region.

# Primary Question
## When is the best time to sell a ticket?
### Deciding when to sell your ticket is one of the most important questions to answer. Being able to view changes in minimum price (get-in price) and ticket count as the event comes closer can be used to make pricing decisions that will help maximize profit.
#### SQL features used: VIEW, CTE, Window function

In [10]:
%%sql
CREATE VIEW charli_xcx_events AS
    SELECT *
    FROM vividseats
    WHERE name = 'Charli XCX'
ORDER BY eventDate;

 * mysql://admin:***@tickets.ceaw7c57s0vu.us-east-1.rds.amazonaws.com/tickets
0 rows affected.


[]

In [11]:
%%sql
WITH charli_xcx_boston AS (
    SELECT
        name,
        venue,
        eventDate - reportDate AS Days_Until_Event,
        ticketCount,
        LAG(ticketCount, 1) OVER(
            ORDER BY reportDate
        ) AS previous_ticket_count,
        (
            (ticketCount - LAG(ticketCount, 1) OVER(ORDER BY reportDate))
            /
            LAG(ticketCount, 1) OVER(ORDER BY reportDate)
        ) * 100 AS percent_change_ticket_count,
        minPrice,
        LAG(minPrice, 1) OVER(
            ORDER BY reportDate
        ) AS previous_min_price,
        (
            (minPrice - LAG(minPrice, 1) OVER(ORDER BY reportDate))
            /
            LAG(minPrice, 1) OVER(ORDER BY reportDate)
        ) * 100 AS percent_change_min_price
    FROM charli_xcx_events
    WHERE vs_event_id = '3768211'
)
SELECT
    name,
    venue,
    Days_Until_Event,
    ticketCount,
    ROUND(percent_change_ticket_count, 2) AS percent_change_ticket_count,
    minPrice,
    ROUND(percent_change_min_price, 2) AS percent_change_min_price
FROM charli_xcx_boston;

 * mysql://admin:***@tickets.ceaw7c57s0vu.us-east-1.rds.amazonaws.com/tickets
9 rows affected.


[('Charli XCX', 'House of Blues Boston', 12, 116, None, 115, None),
 ('Charli XCX', 'House of Blues Boston', 9, 104, Decimal('-10.34'), 125, Decimal('8.70')),
 ('Charli XCX', 'House of Blues Boston', 8, 98, Decimal('-5.77'), 138, Decimal('10.40')),
 ('Charli XCX', 'House of Blues Boston', 8, 102, Decimal('4.08'), 138, Decimal('0.00')),
 ('Charli XCX', 'House of Blues Boston', 6, 100, Decimal('-1.96'), 123, Decimal('-10.87')),
 ('Charli XCX', 'House of Blues Boston', 4, 82, Decimal('-18.00'), 96, Decimal('-21.95')),
 ('Charli XCX', 'House of Blues Boston', 3, 67, Decimal('-18.29'), 103, Decimal('7.29')),
 ('Charli XCX', 'House of Blues Boston', 2, 59, Decimal('-11.94'), 98, Decimal('-4.85')),
 ('Charli XCX', 'House of Blues Boston', 0, 34, Decimal('-42.37'), 73, Decimal('-25.51'))]

### From about two weeks out to one week out, we see a decrease in ticket count and an increase in the minimum price. With 8 days until the event, it reached as high as 138. However, from this point on until the event we see an even greater decrease in ticket count but we also observe the minimum price decreasing as well. Based on these results, it seems like selling a week out from the day of the event would have netted you the most profit. This information can be used in the future if Charli XCX tours again and plays in a market similar to Boston.

# Secondary Questions
## Was this event profitable?
### Another important question that you have to ask yourself is whether or not you should buy an event. Instead of blindly making a 50/50 call, you can rely on historical data to make a better informed decision. This query compares the average minimum resale price from Vividseats to the minimum face value price from Ticketmaster to determine whether an event was profitable.
#### SQL features used: GROUP BY, JOIN, CASE

In [16]:
%%sql
SELECT
    vi.name,
    t.genre,
    vi.venue,
    ROUND(AVG(vi.minPrice)) AS average_resale_get_in,
    t.minPrice AS retail_get_in,
    CASE WHEN AVG(vi.minPrice) > t.minPrice THEN 1 ELSE 0 END AS profitable
FROM vividseats vi
JOIN venue ve
    ON vi.venue = ve.name
JOIN ticketmaster t
    ON ve.name = t.venue
GROUP BY vs_event_id
ORDER BY vi.eventDate, vi.name, reportDate;

 * mysql://admin:***@tickets.ceaw7c57s0vu.us-east-1.rds.amazonaws.com/tickets
35 rows affected.


[('Bon Jovi', 'Country', 'Enmarket Arena', Decimal('52'), 56, 0),
 ('Jawbreaker', 'Pop', 'House of Blues Chicago', Decimal('11'), 28, 0),
 ('Bon Jovi', 'Rock', 'Moody Center ATX', Decimal('106'), 60, 1),
 ('Netflix Is A Joke Fest - Bill Burr', 'Comedy', 'Hollywood Palladium', Decimal('66'), 64, 1),
 ('Clinton Kane', 'Pop', 'The Bronze Peacock at House of Blues Houston', Decimal('45'), 25, 1),
 ('Paul McCartney', 'Rock', 'Oakland Arena', Decimal('106'), 30, 1),
 ('Pearl Jam (Rescheduled from 4/15/2020)', 'Comedy', 'Kia Forum', Decimal('116'), 60, 1),
 ('Garth Brooks', 'Country', 'Notre Dame Stadium', Decimal('78'), 92, 0),
 ('iHeartCountry Festival - Thomas Rhett, Carrie Underwood, Zac Brown Band', 'Rock', 'Moody Center ATX', Decimal('109'), 60, 1),
 ('Mxmtoon', 'Pop', 'House of Blues Chicago', Decimal('8'), 28, 0),
 ('Pearl Jam (Rescheduled from 4/16/2020)', 'Comedy', 'Kia Forum', Decimal('129'), 60, 1),
 ('Dave Matthews Band', 'Rock', 'Moody Center ATX', Decimal('67'), 60, 1),
 ('Dave Matthews Band', 'Rock', 'Dos Equis Pavilion', Decimal('74'), 50, 1),
 ('Garth Brooks', 'Country', 'Paul Brown Stadium', Decimal('77'), 85, 0),
 ('Twice', 'Pop', 'Banc of California Stadium', Decimal('119'), 60, 1),
 ('Dave Matthews Band', 'Rock', 'Brandon Amphitheater', Decimal('91'), 50, 1),
 ('Paul McCartney', 'Rock', 'Dickies Arena', Decimal('404'), 40, 1),
 ('Rex Orange County', 'Rock', 'Germania Insurance Amphitheater', Decimal('55'), 30, 1),
 ('Garth Brooks', 'Country', 'Gillette Stadium', Decimal('47'), 85, 0),
 ('Kane Brown', 'Country', 'Amalie Arena', Decimal('62'), 47, 1),
 ('Luke Combs', 'Country', 'Empower Field at Mile High', Decimal('76'), 21, 1),
 ('Nine Inch Nails', 'Rock', 'The Met Philadelphia', Decimal('86'), 50, 1),
 ('Kane Brown', 'Country', 'Enmarket Arena', Decimal('49'), 56, 0),
 ('Keshi', 'Other', 'Theatre of Living Arts', Decimal('102'), 30, 1),
 ('Coldplay', 'Rock', 'Soldier Field', Decimal('77'), 34, 1),
 ('Paul McCartney', 'Rock', 'Thompson-Boling Arena', Decimal('138'), 70, 1),
 ('Keshi', 'Pop', 'House of Blues Chicago', Decimal('90'), 28, 1),
 ('Morgan Wallen', 'Country', 'PNC Music Pavilion', Decimal('282'), 60, 1),
 ('Coldplay', 'Rock', 'MetLife Stadium', Decimal('72'), 40, 1),
 ('Chris Rock', 'Comedy', 'Taft Theatre', Decimal('62'), 50, 1),
 ('Jimmy Buffett', 'Rock', 'Moody Center ATX', Decimal('18'), 60, 0),
 ('Paul McCartney', 'Rock', 'MetLife Stadium', Decimal('107'), 40, 1),
 ('Rex Orange County', 'Rock', 'BMO Harris Pavilion', Decimal('44'), 30, 1),
 ('Keshi', 'Other', 'The Observatory North Park', Decimal('94'), 30, 1),
 ('Rex Orange County', 'Rock', 'Huntington Bank Pavilion at Northerly Island', Decimal('50'), 30, 1)]

### These results can be used in the future to make better informed buying decisions. For instance, the average resale get in price (106) for Paul McCartney's show at Oakland Arena is a lot higher than the retail get in price (30). If he were to ever play a show again at that venue or in that area (SF Bay Area) I would feel confident buying that show based on how his Oakland Arena show is doing.

## Can we predict profitability based on genre?
### This query builds off of the previous one. To possibly help make our decision-making process easier, we can calculate the percentage of profitable events based on genre.
#### SQL features used: GROUP BY, JOIN, CASE, CTE

In [21]:
%%sql
WITH profitable AS (
    SELECT
        vi.name,
        t.genre,
        vi.venue,
        ROUND(AVG(vi.minPrice)) AS average_resale_get_in,
        t.minPrice AS retail_get_in,
        CASE WHEN AVG(vi.minPrice) > t.minPrice THEN 1 ELSE 0 END AS profitable
    FROM vividseats vi
    JOIN venue ve
        ON vi.venue = ve.name
    JOIN ticketmaster t
        ON ve.name = t.venue
    GROUP BY vs_event_id
    ORDER BY vi.eventDate, vi.name, reportDate
),
profitable_percentage AS (
    SELECT
        genre,
        (((COUNT(CASE WHEN profitable = 1 AND genre = 'Other' THEN profitable ELSE NULL END)) / COUNT(genre)) * 100) AS profitable_other,
        ((COUNT(CASE WHEN profitable = 1 AND genre = 'Comedy' THEN profitable ELSE NULL END)) / COUNT(genre)) * 100 AS profitable_comedy,
        ((COUNT(CASE WHEN profitable = 1 AND genre = 'Pop' THEN profitable ELSE NULL END)) / COUNT(genre)) * 100 AS profitable_pop,
        ((COUNT(CASE WHEN profitable = 1 AND genre = 'Country' THEN profitable ELSE NULL END)) / COUNT(genre)) * 100 AS profitable_country,
        ((COUNT(CASE WHEN profitable = 1 AND genre = 'Rock' THEN profitable ELSE NULL END)) / COUNT(genre)) * 100 AS profitable_rock
    FROM profitable
    GROUP BY genre
)
SELECT
    genre,
    FORMAT(profitable_other, 2) AS other_profitability,
    FORMAT(profitable_comedy, 2) AS comdey_profitability,
    FORMAT(profitable_pop, 2) AS pop_profitability,
    FORMAT(profitable_country, 2) AS country_profitability,
    FORMAT(profitable_rock, 2) AS rock_profitability
FROM profitable_percentage;

 * mysql://admin:***@tickets.ceaw7c57s0vu.us-east-1.rds.amazonaws.com/tickets
5 rows affected.


[('Country', '0.00', '0.00', '0.00', '37.50', '0.00'),
 ('Pop', '0.00', '0.00', '60.00', '0.00', '0.00'),
 ('Rock', '0.00', '0.00', '0.00', '0.00', '93.75'),
 ('Comedy', '0.00', '100.00', '0.00', '0.00', '0.00'),
 ('Other', '100.00', '0.00', '0.00', '0.00', '0.00')]

### The results of this query tells us that country and pop events are not as profitable as rock and comedy. Of course, we are working with a very limited sample of data, but I can see how this query can prove to be more useful and reliable in the future if more events are added to the database.